File: EventStreamManager.Infrastructure/Models/Interface/InterfaceConfig.cs
using System.ComponentModel.DataAnnotations;

namespace EventStreamManager.Infrastructure.Models.Interface;

public class InterfaceConfig
{
    public string Id { get; set; } = string.Empty;
    
    [Required(ErrorMessage = "配置名称不能为空")]
    public string Name { get; set; } = string.Empty;
   
    [Required(ErrorMessage = "请至少选择一个关联的处理器")]
    [MinLength(1, ErrorMessage = "请至少选择一个关联的处理器")]
    public List<string> ProcessorIds { get; set; } = new();
    public List<string> ProcessorNames { get; set; } = new();
    
    
    [Required(ErrorMessage = "接口URL不能为空")]
    [Url(ErrorMessage = "请输入有效的URL地址")]
    [StringLength(500, ErrorMessage = "URL长度不能超过500个字符")]
    public string Url { get; set; } = string.Empty;
    
    [Required(ErrorMessage = "请求方法不能为空")]
    [RegularExpression("^(GET|POST|PUT|DELETE|PATCH|HEAD|OPTIONS)$", 
        ErrorMessage = "请求方法必须是 GET、POST、PUT、DELETE、PATCH、HEAD 或 OPTIONS")]
    public string Method { get; set; } = "POST";
    public List<HeaderItem> Headers { get; set; } = new();
    
    [Range(1, 300, ErrorMessage = "超时时间必须在1-300秒之间")]
    public int Timeout { get; set; } = 30;
    
    
    [Range(0, 10, ErrorMessage = "重试次数必须在0-10之间")]
    public int RetryCount { get; set; } = 3;
  
    [Range(1, 60, ErrorMessage = "重试间隔必须在1-60秒之间")]
    public int RetryInterval { get; set; } = 5;
    public bool Enabled { get; set; }
    
    [Required(ErrorMessage = "请求模板不能为空")]
    public string RequestTemplate { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
}

File: EventStreamManager-UI/src/types/interface-config.ts
/**
 * 接口发送配置相关类型定义
 */

/**
 * Header 项
 */
export interface HeaderItem {
  key: string;
  value: string;
}

/**
 * 接口配置
 */
export interface InterfaceConfig {
  id: string;
  name: string;
  processorIds: string[];
  processorNames: string[];
  url: string;
  method: 'GET' | 'POST' | 'PUT' | 'DELETE';
  headers: HeaderItem[];
  timeout: number;
  retryCount: number;
  retryInterval: number;
  enabled: boolean;
  requestTemplate: string;
  description: string;
}

/**
 * 可用处理器
 */
export interface AvailableProcessor {
  id: string;
  name: string;
}

/**
 * 数据库类型信息（带激活配置）
 */
export interface DatabaseTypeWithActiveConfig {
  value: string;
  label: string;
  activeConfig: {
    id: string;
    name: string;
    connectionString: string;
    driver: string;
    isActive: boolean;
    timeout: number;
  } | null;
}

/**
 * 调试日志条目
 */
export interface DebugLogEntry {
  type: 'info' | 'warn' | 'error' | 'success' | 'output';
  message: string;
  timestamp: string;
}

/**
 * 处理器执行结果
 */
export interface ProcessResultDto {
  needToSend: boolean;
  reason?: string;
  data?: any;
  [key: string]: any;
}

/**
 * 请求信息
 */
export interface RequestInfo {
  url: string;
  method: string;
  headers: Record<string, string>;
  body?: string;
}

/**
 * 响应信息
 */
export interface ResponseInfo {
  statusCode: number;
  headers?: Record<string, string>;
  body?: string;
}

/**
 * 接口调试请求参数
 */
export interface InterfaceDebugRequest {
  interfaceConfigId: string;
  processorId: string;
  databaseType: string;
  eventCode?: string;
  eventId?: string;
}

/**
 * 接口调试响应
 */
export interface InterfaceDebugResponse {
  success: boolean;
  errorMessage?: string;
  logs: DebugLogEntry[];
  executionTimeMs: number;
  processorExecutionTime?: number;
  interfaceExecutionTime?: number;
  processorResult?: ProcessResultDto;
  requestInfo?: RequestInfo;
  responseInfo?: ResponseInfo;
}

/**
 * 接口调试结果（兼容旧版）
 */
export interface InterfaceDebugResult {
  success: boolean;
  requestUrl: string;
  requestMethod: string;
  requestHeaders: Record<string, string>;
  requestBody: string;
  responseStatus?: number;
  responseBody?: string;
  executionTimeMs: number;
  errorMessage?: string;
  processorExecutionTime?: number;
  processorResult?: any;
}